sdk/messaging/azeventgrid/autorest.md

```diff
@@ -20,34 +18,110 @@ use: "@autorest/go@4.0.0-preview.52"
+  - from: 
+    - client.go
+    where: $
+    transform: | 
+      return $.replace(
+        /(func \(client \*Client\) publishCloudEventCreateRequest.+?)return req, nil/s, 
+        '$1\nreq.Raw().Header.Set("Content-type", "application/cloudevents+json; charset=utf-8")\nreturn req, nil');        
+```
+
+Fix the error type so it's a bit more presentable, and looks like an error for this package.
+
+```yaml
+directive:
+  - from: 
+    - models_serde.go
+    - models.go
+    where: $
+    transform: | 
+      return $
+        // remove some types that were generated to support the recursive error.
+        .replace(/\/\/ AzureCoreFoundationsInnerErrorInnererror.+?\n}/s, "")
+        // also, remove its marshalling functions
+        .replace(/\/\/ (Unmarshal|Marshal)JSON implements[^\n]+?for type AzureCoreFoundationsInnerErrorInnererror.+?\n}/sg, "")
+        .replace(/\/\/ AzureCoreFoundationsErrorInnererror.+?\n}/s, "")
+        .replace(/\/\/ (Unmarshal|Marshal)JSON implements[^\n]+?for type AzureCoreFoundationsErrorInnererror.+?\n}/sg, "")
+        .replace(/\/\/ AzureCoreFoundationsErrorResponseError.+?\n}/s, "")
+        .replace(/\/\/ (Unmarshal|Marshal)JSON implements[^\n]+?for type AzureCoreFoundationsErrorResponseError.+?\n}/sg, "")
+        .replace(/\/\/ AzureCoreFoundationsErrorResponse.+?\n}/s, "")
+        .replace(/\/\/ (Unmarshal|Marshal)JSON implements[^\n]+?for type AzureCoreFoundationsErrorResponse.+?\n}/sg, "")
+
+        // Remove any references to the type and replace them with InnerError.
+        //.replace(/Innererror \*(AzureCoreFoundationsInnerErrorInnererror|AzureCoreFoundationsErrorInnererror)/g, "InnerError *InnerError")
+
+        // Fix the marshallers/unmarshallers to use the right case.
+        .replace(/(a|c).Innererror/g, '$1.InnerError')
+        .replace(/Innererror \*AzureCoreFoundationsInnerError/g, "InnerError *InnerError")
+
+        // We have two "inner error" types that are identical (ErrorInnerError and InnerError). Let's eliminate the one that's not actually directly referenced.
+        .replace(/\/\/azureCoreFoundationsInnerError.+?\n}/s, "")
+        
+        //
+        // Fix the AzureCoreFoundation naming to match our style.
+        //
+        .replace(/AzureCoreFoundations/g, "")
+```
+
+Trim out the 'Interface any' for types that are empty.
+
+```yaml
+directive:
+  - from: response_types.go
     where: $.definitions.CloudEvent.properties.specversion
-    transform: $["x-ms-client-name"] = "SpecVersion"
-  - from: swagger-document
-    where: $.definitions.CloudEvent.properties.datacontenttype
-    transform: $["x-ms-client-name"] = "DataContentType"
-  - from: swagger-document
-    where: $.definitions.CloudEvent.properties.dataschema
-    transform: $["x-ms-client-name"] = "DataSchema"
-  # mark models as external so they're just omitted
+    transform: $.replace(/\s+\/\/ Anything\s+Interface any/sg, "$1");
```

Comment:
Does this actually work? The `where` clause is for swagger but the `from` is Go source code.

Reply:
It does but apparently `where` is just ignored for source transforms. Or...it's used but doesn't affect the transform itself.

I cleaned it up :)